lec15-data-viz.Rmd

```diff
@@ -1,4 +1,333 @@
+>
+
+
+There are many considerations to take into account when creating a graph.
+
+Data types:
+* Nominal (labels without quantitative values)
+* Ordinal (the order of the values is what’s important, but the magnitude of difference between each value is not really known)
+* Interval (interval scales are numeric scales in which we know not only the order, but also the exact differences between the values)
+* Ratio (same properties as interval scales but with an absolute zero, e.g. height)
+
+And data features:
+* Scalar (simple magnitude or quantity)
+* Vector (magnitude and direction)
+* Tensor (vectors + other attributes)
+
+Ask: "what kind of data/information is it?"
+* What type?
+* Numerical?
+* Spatial?
+* How many dimensions?
+* Nominal/categorical?
+* Narrative?
+
+## Tufte's guidelines
+* Reduce non-data ink
+* Enhance the data ink
+
+Reduces the proportion of graphic’s ink devoted to the non-redundant display of data-information,
+
+Avoid "chartjunk" - extraneous visual elements that detract from message.
+
+Visual Attribute    | Non-Data | Data
+--------------------|----------|-------
+Line Thickness      | Thin     | Thick
+Size                | Small    | Big
+Brightness Contrast | Decrease | Increase
+Colour Saturation   | Decrease | Increase
+
+## Colour rules
+* Large background colours should be quiet, muted to let brighter colors stand out
+* To highlight some element of a figure, using a bright colour can be effective
+* However, if brightly colouring this aspect of your figure serves no purpose, leave it greyscale/plain
+* Here is a reference sheet with all the [colours and colour names in R](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf)
+
+
+## Summary
+* Determine what type of data you have and how your data will be used
+* Determine how many variable types are in your data
+* Decide on a visual treatment for each of the variables 
+* Focus on the data in your visual representation
+
+## More general resources on plotting
+* [Ten Simple Rules for Better Figures](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003833)
+* [Finding the Right Color Palettes for Data Visualizations](https://blog.graphiq.com/finding-the-right-color-palettes-for-data-visualizations-fcd4e707a283)
+* [Cookbook for R - Graphs](http://www.cookbook-r.com/Graphs/)
+* [ggplot cheatsheet - very useful!](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf)
+
+## What is ggplot?
+ggplot2 is meant to be an implementation of the Grammar of Graphics, hence gg-plot. The basic notion is that there is a grammar to the composition of graphical components in statistical graphics, and by directly controlling that grammar, you can generate a large set of carefully constructed graphics tailored to your particular needs. Each component is added to the plot as a layer.
+Here is the [ggplot book](https://www.cs.uic.edu/~wilkinson/TheGrammarOfGraphics/GOG.html).
```

Comment:
As far as I understand, this book actually does not have anything to do with the R-packages `ggplot`/`ggplot2`. It appears to be a more theoretical and its principles inspired the development of the R-packages. So remove this sentence and just include the link in the first sentence of the previous paragraph.

Change

```
ggplot2 is meant to be an implementation of the Grammar of Graphics, hence gg-plot
``` 

to something like

```
ggplot2 is inspired by the principles from the [Grammar of Graphics](https://www.cs.uic.edu/~wilkinson/TheGrammarOfGraphics/GOG.html) and also named thereafter.
```

Reply:
very good, I used that verbatim